src/classes/Store.ts

```diff
@@ -0,0 +1,93 @@
+import { assert } from "chai";
+import equal from "fast-deep-equal/es6";
+import Entity from "./Entity";
+
+export default class Store {
+  private state: Map<string, Entity> = new Map();
+
+  public hydrateWithEntities = (entities: Map<string, Entity>) => {
+    assert(
+      this.state.size === 0,
+      "State is not empty. Please use state.clear() to clear the state before hydrating.",
+    );
+    this.state = entities;
+  }
+
+  public hydrateWithJson = (stringifiedJson: string) => {
+    assert(
+      this.state.size === 0,
+      "State is not empty. Please use state.clear() to clear the state before hydrating.",
+    );
+    assert(stringifiedJson.trim() !== "", "JSON cannot be an empty string.");
+    this.state = new Map(JSON.parse(stringifiedJson));
+  }
+
+  public readState = () => {
+    return JSON.stringify(Array.from(this.state.entries()));
+  }
+
+  public addEntity = (entityKey: string, entity: Entity): void => {
```

Comment:
addEntity replaces if an event with the same key already exists? maybe we should throw an error in addEntity if it already exists, but have a separate method setEntity to be able to replace if it does not exists?